tls/s2n_server_key_exchange.c

```diff
@@ -58,9 +58,7 @@ static int s2n_ecdhe_server_key_recv(struct s2n_connection *conn)
     if (conn->actual_protocol_version == S2N_TLS12) {
         s2n_hash_algorithm hash_algorithm;
         s2n_signature_algorithm signature_algorithm;
-        int matched = s2n_get_signature_hash_pair_if_supported(in, &hash_algorithm, &signature_algorithm);
-
-        S2N_ERROR_IF(!matched, S2N_ERR_BAD_MESSAGE);
+        GUARD(s2n_get_signature_hash_pair_if_supported(in, &hash_algorithm, &signature_algorithm));
```

Comment:
Good catch. Are there tests that check this behavior? I'm surprised a test didn't fail somewhere down the line if there wasn't a match and `hash_algorithm` wasn't set in `s2n_get_signature_hash_pair_if_supported`.

Reply:
I didn't find any and it's a bit tricky to reproduce, as servers (at least that was the case with openssl) will fail handshake after receiving client hello if there is no match for sigalgs.